src/app.jsx

```diff
@@ -36,12 +36,43 @@ import {
     usageStartPolling,
     usageStopPolling,
 } from "./libvirt-dbus.js";
+import { useObject, useEvent } from "hooks";
+import * as service from 'service.js';
 
+import {
+    updateLibvirtState,
+} from './actions/store-actions.js';
 const _ = cockpit.gettext;
 
 superuser.reload_page_on_change();
 
-class App extends React.Component {
+export const App = ({ store, name }) => {
+    const libvirtService = useObject(() => service.proxy(name), null, []);
+
+    useEvent(libvirtService, "changed", () => {
+        store.dispatch(updateLibvirtState({
+            activeState: libvirtService.exists ? libvirtService.state : 'stopped',
```

Comment:
is this special case necessary? `libvirtService.state === undefined` for a non-existing service, which may work just as well. There is a lot of states, so I suppose the code can't rely on an exhaustive comparison anyway? The code below only checks for `!= running` anyway.

Reply:
We indeed don't need this special case, will remove.